pkg/controller/statusmanager/status_manager.go

```diff
@@ -186,6 +190,33 @@ func (status *StatusManager) SetDeployments(deployments []types.NamespacedName)
 func (status *StatusManager) SetRelatedObjects(relatedObjects []configv1.ObjectReference) {
 	status.Lock()
 	defer status.Unlock()
+
+	var found bool = false
+	for _, status_obj := range status.relatedObjects {
+		for _, obj := range relatedObjects {
+			found = reflect.DeepEqual(status_obj, obj)
+
+			if found {
+				break
+			}
+		}
+		if !found {
+			gvr := schema.GroupVersionResource{
+				Group:    status_obj.Group,
+				Resource: status_obj.Resource,
+			}
+			gvk, err := status.mapper.KindFor(gvr)
+			if err != nil {
+				log.Printf("Error getting GVK of object for deletion: %v", err)
+				continue
+			}
+			obj_to_delete := &uns.Unstructured{}
+			obj_to_delete.SetName(status_obj.Name)
+			obj_to_delete.SetNamespace(status_obj.Namespace)
+			obj_to_delete.SetGroupVersionKind(gvk)
+			status.client.Delete(context.TODO(), obj_to_delete)
```

Comment:
log an error on error

Reply:
If possible, we want to do a Background delete here. That means that we won't wait for descendant objects (e.g. pods of deployments) to delete - so we don't block the operator.